lib/rules/no-inner-declarations.js

```diff
@@ -97,7 +111,15 @@ module.exports = {
 
         return {
 
-            FunctionDeclaration: check,
+            FunctionDeclaration(node) {
+                const isStrictFunction = sourceCode.getScope(node).isStrict;
+
+                if (isStrictFunction && blockScopedFunctions === "allow" && es6EcmaVersionsList.has(ecmaVersion)) {
```

Comment:
```suggestion
                if (isStrictFunction && blockScopedFunctions === "allow" && ecmaVersion >= 2015) {
```

We can simplify this.